packages/bot-web-ui/src/app.jsx

```diff
@@ -37,19 +49,24 @@ class App extends React.Component {
 
     render() {
         return (
-            <MobxContentProvider store={this.root_store}>
-                <div className='bot'>
-                    <BotNotificationMessages />
-                    <Toolbar />
-                    <MainContent />
-                    <RunPanel />
-                    <QuickStrategy />
-                    <BotFooterExtensions />
-                    <Audio />
-                    <RoutePromptDialog />
-                    <BlocklyLoading />
-                </div>
-            </MobxContentProvider>
+            <>
+                {this.state.is_loading && <Loading />}
+                {!this.state.is_loading && (
+                    <MobxContentProvider store={this.root_store}>
+                        <div className='bot'>
+                            <BotNotificationMessages />
+                            <Toolbar />
+                            <MainContent />
+                            <RunPanel />
+                            <QuickStrategy />
+                            <BotFooterExtensions />
+                            <Audio />
+                            <RoutePromptDialog />
+                            <BlocklyLoading />
+                        </div>
+                    </MobxContentProvider>
+                )}
+            </>
```

Comment:
```suggestion
        if (this.state.is_loading) return <Loading />;
        return (
                    <MobxContentProvider store={this.root_store}>
                        <div className='bot'>
                            <BotNotificationMessages />
                            <Toolbar />
                            <MainContent />
                            <RunPanel />
                            <QuickStrategy />
                            <BotFooterExtensions />
                            <Audio />
                            <RoutePromptDialog />
                            <BlocklyLoading />
                        </div>
                    </MobxContentProvider>
                );
```

Reply:
@salar-fs I wanted to do that, but we have a line in our documentation which said `Use `{condition && <el/> ...</el>}` syntax to conditionally render an element.`, not sure should it apply here or not?

Reply:
I think it depends on the situation. IMO here suggested change really simplifies logic, removes the need to check `is_loading` twice and improves code readability.